incubator/virtualcluster/experiment/pkg/scheduler/cache/cache.go

```diff
@@ -334,3 +334,19 @@ func (c *schedulerCache) RemoveProvision(clustername, key string) error {
 	}
 	return clusterState.RemoveProvision(key)
 }
+
+func (c *schedulerCache) Dump() string {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	out := "-- Dump Super Clusters --"
+	for k, v := range c.clusters {
+		out = fmt.Sprintf("%s\n%s: %s", out, k, v.Dump())
```

Comment:
try using `strings.Builder` for better performance in this case
builder.Write([]byte)